mono/metadata/threads.c

```diff
@@ -670,12 +670,16 @@ mono_thread_internal_set_priority (MonoInternalThread *internal, MonoThreadPrior
 
 #ifdef HOST_WIN32
 	BOOL res;
+	DWORD last_error;
 
 	g_assert (internal->native_handle);
 
+	MONO_ENTER_GC_SAFE;
 	res = SetThreadPriority (internal->native_handle, priority - 2);
+	last_error = GetLastError ();
```

Comment:
One could avoid the call if res, but it is incredibly cheap, so ok.
SetThreadPriority is a "slow" syscall, GetLastError is not.

We should have ```inline int-or-enum  MonoToWin32ThreadPriority(int-or-enum a) { return a - 2;}```

I added a comment about the `2` somewhere, probably in `cxx`.